autogpt_platform/backend/backend/executor/manager.py

```diff
@@ -720,6 +711,8 @@ def cleanup(self):
 
     @property
     def agent_server_client(self) -> "AgentServer":
+        # Since every single usage of this property happens from a different thread,
+        # there is no value in caching it.
         return get_agent_server_client()
```

Comment:
I put that comment there to prevent future confusion and time wasted on this. I tried to make the property cached because it seemed like a waste to re-create the client for every single use. Then I found out that that's illegal and will cause errors because a client can only be owned by one thread. I reverted the technical change and left a note for the next person who might think it's a good idea to re-use the client.